filters.go

```diff
@@ -134,3 +134,26 @@ func (p *Pipe) Concat() *Pipe {
 	})
 	return p.WithReader(io.MultiReader(readers...))
 }
+
+// First reads from the pipe, and returns a new pipe containing only the first N
+// lines. If there is an error reading the pipe, the pipe's error status is also
+// set.
+func (p *Pipe) First(lines int) *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	scanner := bufio.NewScanner(p.Reader)
+	output := strings.Builder{}
+	for i := 0; i < lines; i++ {
+		if !scanner.Scan() {
+			break
+		}
+		output.WriteString(scanner.Text())
+		output.WriteRune('\n')
+	}
+	err := scanner.Err()
+	if err != nil {
+		p.SetError(err)
+	}
```

Comment:
How does this implementation behave, if you pass a very big file and you use the `First()` filter? With the unix commands, the pipe is closed by `head` after 10 lines, which allows all the commands in front of `head` in the pipe to end as well, without the need of reading the whole file.
Would this work here as well, if we would close the pipe?

Reply:
That is a very good point. We don't read any more of the pipe than we need to, but we also don't close it. I'd better add that...

Reply:
Done in 8921a7d